### PR TITLE
New: Kavita Connection

### PR DESCRIPTION
**Pull request by majora2007**
#### Database Migration
NO

#### Description
Adds a new Connector for [Kavita](https://github.com/Kareadita/Kavita), a self-hosted book reader, so that when Readarr manages files, Kavita can be told via notification and consume these changes. Note: Kavita must be v0.5.6 or greater for this to work.

#### Screenshot (if UI related)
image

#### Todos
- [ ] Tests
- [ ] Translation Keys (./src/NzbDrone.Core/Localization/Core/en.json)
- [ ] [Wiki Updates](https://wiki.servarr.com)

Please let me know if Localization is needed. I mainly based on Prowl and Subsonic, both of which do not have Localization nor Tests.

### CODE REVIEW

File: src/NzbDrone.Core/Notifications/Kavita/KavitaSettings.cs [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+    [FieldDefinition(1, Label = "Port")]
+    public int Port { get; set; }
+
+    [FieldDefinition(3, Label = "API Key", Privacy = PrivacyLevel.ApiKey, HelpLink = "https://wiki.kavitareader.com/en/guides/settings/opds")]
+    public string ApiKey { get; set; }
+
+    [FieldDefinition(4, Label = "Use SSL", Type = FieldType.Checkbox, HelpText = "Connect to Kavita over HTTPS instead of HTTP")]
+    public bool UseSsl { get; set; }
+
+    [FieldDefinition(5, Label = "Have Kavita kick off a scan", Type = FieldType.Checkbox)]
```
Comment (Qstick) [inline src/NzbDrone.Core/Notifications/Kavita/KavitaSettings.cs]: Can we make `Have Kavita kick off a scan` the HelpText and simplify the label? Something like `Scan` or `Update Library`. This should keep consistency with other library updates (Plex, Kodi, etc)

File: src/NzbDrone.Core/Notifications/Kavita/KavitaSettings.cs [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+        Port = 4040;
+    }
+
+    [FieldDefinition(0, Label = "Host")]
+    public string Host { get; set; }
+
+    [FieldDefinition(1, Label = "Port")]
+    public int Port { get; set; }
+
+    [FieldDefinition(3, Label = "API Key", Privacy = PrivacyLevel.ApiKey, HelpLink = "https://wiki.kavitareader.com/en/guides/settings/opds")]
```
Comment (Qstick) [inline src/NzbDrone.Core/Notifications/Kavita/KavitaSettings.cs]: Field number skips from 1 to 3

File: src/NzbDrone.Core/Notifications/Kavita/Kavita.cs [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+    public override void OnBookDelete(BookDeleteMessage deleteMessage)
+    {
+        var allPaths = deleteMessage.Book.BookFiles.Value.Select(v => v.Path).Distinct();
+        var path = Directory.GetParent(allPaths.First())?.FullName;
+        Notify(Settings, "Readarr - Book Deleted", path);
+    }
+
+    public override void OnBookFileDelete(BookFileDeleteMessage message)
+    {
+        Notify(Settings, "Readarr - Book File Deleted", Directory.GetParent(message.BookFile.Path)?.FullName);
```
Comment (Qstick) [inline src/NzbDrone.Core/Notifications/Kavita/Kavita.cs]: Do we have pre-defined variables for these? Similar to what you used at L45 (BOOK_RETAGGED_TITLE_BRANDED)

File: src/NzbDrone.Core/Notifications/Kavita/Kavita.cs [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+        var allPaths = message.BookFiles.Select(v => v.Path).Distinct();
+        var path = Directory.GetParent(allPaths.First())?.FullName;
+        Notify(Settings, "Readarr - Downloaded", path);
+    }
+
+    public override void OnBookDelete(BookDeleteMessage deleteMessage)
+    {
+        var allPaths = deleteMessage.Book.BookFiles.Value.Select(v => v.Path).Distinct();
+        var path = Directory.GetParent(allPaths.First())?.FullName;
+        Notify(Settings, "Readarr - Book Deleted", path);
```
Comment (Qstick) [inline src/NzbDrone.Core/Notifications/Kavita/Kavita.cs]: Same for `Redarr - Book Deleted`

File: src/NzbDrone.Core/Notifications/Kavita/Kavita.cs [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+        _logger = logger;
+    }
+
+    public override string Link => "https://www.kavitareader.com/";
+
+    public override void OnReleaseImport(BookDownloadMessage message)
+    {
+        var allPaths = message.BookFiles.Select(v => v.Path).Distinct();
+        var path = Directory.GetParent(allPaths.First())?.FullName;
+        Notify(Settings, "Readarr - Downloaded", path);
```
Comment (Qstick) [inline src/NzbDrone.Core/Notifications/Kavita/Kavita.cs]: Same for `Readarr - Downlaoded`

**Review by Qstick — CHANGES_REQUESTED** [review]

**Comment by majora2007** [comment]
I see a ton of unit tests are broken, but not sure why. They don't seem related to my code at all. 

Edit: Looks like they are all fixed up.

File: src/NzbDrone.Core/Notifications/Kavita/KavitaSettings.cs (hunk 1 above)
Comment (majora2007) [reply src/NzbDrone.Core/Notifications/Kavita/KavitaSettings.cs]: Sounds good, I copied it from Subsonic, but agree `Update Library` is much cleaner

File: src/NzbDrone.Core/Notifications/Kavita/Kavita.cs (hunk 3 above)
Comment (majora2007) [reply src/NzbDrone.Core/Notifications/Kavita/Kavita.cs]: Got it, will apply across the code.

File: src/NzbDrone.Core/Notifications/Kavita/KavitaSettings.cs (hunk 1 above)
Comment (Qstick) [reply src/NzbDrone.Core/Notifications/Kavita/KavitaSettings.cs]: Thanks, I'll fix that one to match yours

**Review by Qstick — APPROVED** [review]

**Pull request merged**